src/partition/repart.c

```diff
@@ -887,6 +887,29 @@ static int config_parse_label(
         return 0;
 }
 
+static int config_parse_uuid(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        sd_id128_t *id = (sd_id128_t *)(data);
+
+        if (sd_id128_from_string(rvalue, id) != 0){
+                *id = SD_ID128_NULL;
+                log_syntax(unit, LOG_WARNING, filename, line, 0, "Partition UUID not valid, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        return 0;
+}
+
```

Comment:
actually, I think we should just move config_parse_id128() from src/nspawn/nspawn-settings.c to shared/conf-parser.c and then reuse that here. it does the exact same thing.

Reply:
I copied the config_parse_id128 from nspawn over into conf-parser.c and added another error condition: All 0 id/uuid.